lib/backpex/live_resource.ex

```diff
@@ -1215,12 +1217,43 @@ defmodule Backpex.LiveResource do
     Map.get(item, primary_key)
   end
 
+  @doc """
+  Returns the pubsub settings for the current LiveResource.
+  """
+  def pubsub(live_resource) do
+    [
+      server: live_resource.config(:pubsub)[:server] || Application.fetch_env!(:backpex, :pubsub_server),
+      topic: live_resource.config(:pubsub)[:topic] || fallback_pubsub_topic(live_resource),
+      event_prefix: live_resource.config(:pubsub)[:event_prefix] || fallback_pubsub_event_prefix(live_resource)
+    ]
+  end
+
+  defp fallback_pubsub_topic(live_resource) do
+    live_resource
+    |> resource_name()
+    |> Kernel.<>("s")
+  end
+
+  defp fallback_pubsub_event_prefix(live_resource) do
+    live_resource
+    |> resource_name()
+    |> Kernel.<>("_")
+  end
+
+  defp resource_name(live_resource) do
+    live_resource
+    |> Module.split()
+    |> List.last()
+    |> String.trim_trailing("Live")
+    |> String.downcase()
+  end
```

Comment:
```suggestion
  defp resource_name(live_resource) do
    live_resource
    |> Module.split()
    |> List.last()
    |> String.downcase()
    |> String.trim_trailing("live")
  end
```

Still, not sure about this. It could lead to some bugs that are difficult to debug. For example, if you don't stick to the `<Resource>Live` naming convention. Then it can happen that multiple LiveResources generate the same PubSub name (imagine `User.IndexLive` and `Post.IndexLive`).

That's also the reason I suggested more explanation in our guides on how we generate PubSub config values automatically.

Reply:
What are your thoughts on using `singular_name/0` and `plural_name/0`?

Reply:
As discussed: We are going to use the live resource module name as a default channel.